devito/symbolics/manipulation.py

```diff
@@ -248,7 +248,7 @@ def xreplace_indices(exprs, mapper, key=None, only_rhs=False):
         handle = [i for i in handle if i.base.label in key]
     elif callable(key):
         handle = [i for i in handle if key(i)]
-    mapper = dict(zip(handle, [uxreplace(i, mapper) for i in handle]))
+    mapper = dict(zip(handle, [i.xreplace(mapper) for i in handle]))
```

Comment:
why back to xreplace?

Reply:
`xreplace` to be used every time you need a re-evaluation of the arguments. Here, we need it (my bad missing this in the previous PR) . Without xreplace, you get indices such as `x + 3 + 4` instead of `x + 7` which makes the generated code correct, but ugly. To reproduce the issue on master, try running `tti_example.py` and you'll see it in some of the temporary expressions

Reply:
so yeah this has nothing to do with the PR per se. I noticed it, and fixed it.

Reply:
Ok